src/engine/contactsengine.cpp

```diff
@@ -737,8 +769,21 @@ class JobContactReader : public ContactReader
 
 void JobThread::run()
 {
-    ContactsDatabase database;
-    if (!database.open(QString(QLatin1String("qtcontacts-sqlite-job-%1")).arg(m_databaseUuid), m_nonprivileged, true)) {
+    QString dbId(QStringLiteral("qtcontacts-sqlite%1-job-%2"));
+    dbId = dbId.arg(m_autoTest ? QStringLiteral("-test") : QString()).arg(m_databaseUuid);
+
+    QMutexLocker locker(&m_mutex);
+
+    m_database.open(dbId, m_nonprivileged, m_autoTest);
+    m_nonprivileged = m_database.nonprivileged();
+    m_running = true;
+
+    {
+        MutexUnlocker unlocker(locker);
+        m_wait.wakeOne();
+    }
```

Comment:
If you're holding the lock when you wake the waiting thread, it will be woken and then immediately lock up on trying to relock the lock that you're holding, so you get two pointless context switches.

I don't see any advantage to using an additional wait condition, this use precedes the existing uses, and can't be incorrectly invoked again.

QSqlDatabase explicitly documents that you can't transfer the connection between threads.  Even if SQLite allows it to work correctly now, there's nothing to stop a future version of QtSql from breaking it.